tests/framework/extensions/provisioning/ssh.go

```diff
@@ -29,15 +35,46 @@ func CallSSHTestByName(testName string, node *nodes.Node) error {
 			return err
 		}
 		str_output := output[:strings.IndexByte(output, '\n')]
-		logrus.Info("CheckCPU test on node " + node.PublicIPAddress + " | Cluster agent cpu usage is: " + str_output + "%")
+		logrus.Infof("CheckCPU test on node %s | Current cluster agent cpu usage is: %s%%", node.PublicIPAddress, str_output)
 
 		output_int, err := strconv.ParseFloat(strings.TrimSpace(str_output), 32)
 		if output_int > cpuUsageVar {
-			logrus.Warn("Cluster agent cpu usage is too high on node" + node.PublicIPAddress + " | Current cpu usage is: " + str_output + "%")
+			logrus.Warnf("Cluster agent cpu usage is too high on node %s | Current cluster agent cpu usage is: %s%%", node.PublicIPAddress, str_output)
 		}
 		if err != nil {
 			return err
 		}
+	case nodeReboot:
+		logrus.Infof("Running NodeReboot test on node %s", node.PublicIPAddress)
+		command := "sudo reboot"
+		_, err := node.ExecuteCommand(command)
+		if err != nil {
+			// There is going to be an error because there will be no response in the ExecuteCommand func
+			// The error will always be "wait: remote command exited without exit status or exit signal"
+			logrus.Info(err)
+		}
+		time.Sleep(10 * time.Second)
+		err = nodev1.AllMachineReady(client, clusterID)
```

Comment:
Is it possible to poll this instead of a mandatory sleep? Actually looking at the logic for `AllMachineReady`, I don't think `AllMachineReady` should be polling. I think it would make sense to pull the polling out of there, or potentially separate into two separate functions, `AllMachinesReady` and `WaitForAllMachinesReady. The issue with forcing timeouts inside framework functions is that there is no way to opt out, so ideally you want to fail fast if the nodes are not ready within ~1 minute, as opposed to the current 30 minute timeout this test would have. @thaneunsoo @slickwarren WDYT?

Reply:
Then in my opinion let's lower the time out. 30 mins seems much too long as well, or have this function take a timeout. I don't really have a strong opinion. Let me what you all think.

Reply:
Actually thinking about it more, since this func is also used for node scaling then we should have a timeout as a parameter right? That way our logic is still encompassed in a poll, and we can fail fast in this case ...

Reply:
yea, I ran multiple tests for this yesterday and it only took around 5 minutes for the cluster to come back up active.